lib/spack/spack/util/web.py

```diff
@@ -554,7 +554,10 @@ def find_versions_of_archive(
         #   .sha256
         #   .sig
         # However, SourceForge downloads still need to end in '/download'.
-        url_regex += r'(\/download)?$'
+        url_regex += r'(\/download)?'
+        # PyPI adds #sha256=... to the end of the URL
```

Comment:
Not a blocker and I would have made the same choice, but if I extend this to CRAN/CPAN, I think that point would be a good time to break some of this method out by repo.

Reply:
We could make this more generic and use `#...` but there are so many URL formats that I'm not sure if this is a good idea.